src/brevitas/graph/equalize.py

```diff
@@ -1342,7 +1344,10 @@ def _apply_rotate(
         else:
             try:
                 # Build hadamard rotation matrix
-                rot_mat, K = get_hadK(hidden_dim)
+                original_rot_mat, original_k = rot_mat, K = get_hadK(hidden_dim)
```

Comment:
There could be a better way to do this. In general, we generate this "too early", so the idea is to generate both